scripts/flash/dispute-contribute.js

```diff
@@ -91,61 +97,30 @@ function disputeContributeInternal(augur, marketId, outcome, amount, disputerAut
-  console.log(chalk.red("parameter 4: amount of REP is optional, will all of user REP"));
-  console.log(chalk.red("parameter 5: invalid is optional, default is false"));
-  console.log(chalk.yellow("user will be give REP if balance is 0"));
-  callback(null);
-}
-
-function disputeContribute(augur, params, auth, callback) {
-  if (!params || params === "help" || params.split(",").length < 2) {
-    help(callback);
-  } else {
-    console.log(params);
-    var paramArray = params.split(",");
-    var marketId = paramArray[0];
-    var outcomeId = paramArray[1];
-
-    var userAuth = null;
-    if (process.env.REPORTER_PRIVATE_KEY) {
-      userAuth = getPrivateKeyFromString(process.env.REPORTER_PRIVATE_KEY);
-    } else if (paramArray[2] && paramArray[2].length > 0) {
-      userAuth = getPrivateKeyFromString(paramArray[2]);
-    }
-    if (!userAuth) {
-      userAuth = auth;
-    }
-
-    var amount = 0;
-    if (paramArray[3] !== undefined) {
-      amount = paramArray[3];
-    }
-    var invalid = paramArray.length === 5 ? paramArray[4] : false;
-    if (!userAuth) userAuth = auth;
-    console.log(chalk.yellow.dim("marketId"), marketId);
-    console.log(chalk.yellow.dim("outcome"), outcomeId);
-    console.log(chalk.yellow.dim("owner"), auth.address);
-    console.log(chalk.yellow.dim("invalid"), invalid);
-    if (amount === 0) {
-      var universe = augur.contracts.addresses[augur.rpc.getNetworkID()].Universe;
-      getBalance(augur, universe, userAuth.address, function (err, balances) {
-        if (err) {
-          console.log(chalk.red(err));
-          return callback(JSON.stringify(err));
-        }
-        amount = balances.reputation;
-        console.log(chalk.yellow.dim("amount"), amount);
-        disputeContributeInternal(augur, marketId, outcomeId, amount, userAuth, invalid, auth, callback);
-      });
-    } else {
+function disputeContribute(augur, args, auth, callback) {
+  if (args === "help" || args.opt.help) {
+    help();
+    return callback(null);
+  }
+  var amount = args.opt.amount;
+  var userAuth = auth;
+  var marketId = args.opt.marketId;
+  var outcome = args.opt.outcome;
+  var invalid = args.opt.invalid;
+  if (amount == null) {
+    var universe = augur.contracts.addresses[augur.rpc.getNetworkID()].Universe;
+    getBalance(augur, universe, userAuth.address, function (err, balances) {
+      if (err) {
+        console.log(chalk.red(err));
+        return callback(JSON.stringify(err));
+      }
+      amount = balances.reputation;
       console.log(chalk.yellow.dim("amount"), amount);
-      disputeContributeInternal(augur, marketId, outcomeId, amount, userAuth, invalid, auth, callback);
-    }
+      disputeContributeInternal(augur, marketId, outcome, amount, userAuth, invalid, auth, callback);
+    });
+  } else {
+    console.log(chalk.yellow.dim("amount"), amount);
+    disputeContributeInternal(augur, marketId, outcome, amount, userAuth, invalid, auth, callback);
```

Comment:
Recommend moving these 2 lines outside the if/else block and removing the else block.